contracts/bonding/BondingManager.sol

```diff
@@ -884,21 +835,78 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         return delegators[_delegator].unbondingLocks[_unbondingLockId].withdrawRound > 0;
     }
 
+    function increaseTranscoderTotalStake(address _transcoder, uint256 _amount) internal {
+        uint256 newStake = transcoderTotalStake(_transcoder).add(_amount);
+        uint256 currentRound = roundsManager().currentRound();
+        uint256 nextRound = currentRound.add(1);
+
+        // If the transcoder is already in the active set update its stake and return
+        if (transcoderPool.contains(_transcoder)) {
+            transcoderPool.updateKey(_transcoder, newStake, address(0), address(0));
+            nextRoundTotalActiveStake = nextRoundTotalActiveStake.add(_amount);
+            lastActiveStakeUpdateRound[_transcoder] = nextRound;
+        } else {
+            // Check if the transcoder is eligible to join the active set in the update round
+            tryToJoinActiveSet(_transcoder, newStake, currentRound);
+        }
+    }
+
+    function decreaseTranscoderTotalStake(address _transcoder, uint256 _amount) internal {
```

Comment:
Currently, whenever we decrease a delegator's `delegatedAmount` field, we also check if the delegate is a registered transcoder and if so we call `decreaseTranscoderTotalStake()`. But, given how `decreaseTranscoderTotalStake()` is implemented we need to make sure that `delegatedAmount` is decreased *after* `increaseTranscoderTotalStake()` or else we will count `_amount` twice when updating active stake.

I see a few things that are not problems right now, but we should be careful about:

- Making sure to always decrease `delegatedAmount` *after* calling `decreaseTranscoderTotalStake()`
- Making sure to always update `delegatedAmount` when we call `decreaseTranscoderTotalStake()` and vice-versa

One way that we can reduce the chance of error is to:

- Change the name of `decreaseTranscoderTotalStake()` to `decreaseTotalStake()` (I'm going simple with the naming)
- Move the logic for increasing the `delegatedAmount` field on a `Delegator` struct into `decreaseTotalStake()`
- Move the logic for checking if an address is a registered transcoder into this function and execute the current function body only if the address is a registered transcoder

WDYT?